metadata-ingestion/src/datahub/ingestion/source/bigquery_v2/usage.py

```diff
@@ -1046,3 +1094,16 @@ def test_capability(self, project_id: str) -> None:
         for entry in self._get_parsed_bigquery_log_events(project_id, limit=1):
             logger.debug(f"Connection test got one {entry}")
             return
+
+    def report_status(self, step: str, status: bool) -> None:
+        if self.redundant_run_skip_handler:
+            self.redundant_run_skip_handler.report_current_run_status(step, status)
+
+    def update_time_window(self):
+        if self.redundant_run_skip_handler:
+            (
+                self.usage_start_time,
+                self.usage_end_time,
```

Comment:
Can we put these and the other updated times in the report somewhere? We should be able to know when the times were expanded / reduced even when the log is truncated

Reply:
done

Reply:
Its done as below:
1. report config provided start time and end time and whether stateful ingestion is enabled `stateful_usage_ingestion_enabled`, `stateful_lineage_ingestion_enabled` in `window_start_time` and `window_end_time` (`add_report_to_config`) in source's init.
2. report final start time and end time used in lineage/usage extractor after `get_time_window` call as `lineage_start_time`/`usage_start_time`, `lineage_end_time`/`usage_end_time`.

I see, the report doesn't directly say that the time window was expanded/reduced but it can be inferred by comparing config start/end time and final start/end time.

Reply:
Ah I missed the double assignment. Works for me